clients/python-wrapper/lakefs/object.py

```diff
@@ -538,6 +538,8 @@ def _upload_raw(self) -> lakefs_sdk.ObjectStats:
                                                                                                headers=headers,
                                                                                                body=self._fd)
 
+        if self._mode.startswith("x") and resp.status == http.HTTPStatus.PRECONDITION_FAILED:
```

Comment:
You can integrate this check into the _io_exception_handler to return the correct exception